### PR TITLE
executor: fix the pessimistic lock on the temporary test case

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Problem Summary:

The old test case is incorrect.

MVCC API read encounter a pessimistic lock do not report error, it should use RawKV API

https://github.com/pingcap/tidb/compare/master...tiancaiamao:fix-test?expand=1#diff-ff7af5dbd454cd98c804a32d8049cd73e476ad819042c63dbe690502f915afeeL8318-L8320

### What is changed and how it works?

What's Changed:

The test can be changed to use a RawKV API or construct a write conflict with the pessimistic lock.
But RawKV API is not convenient, and construct  a write conflict can't be done use simple SQL here (write on the temporary table ID is not pushed to TiKV) ... 
So I just remove it.

The other test can be used to cover the pessimistic lock on the temporary table do not use the network.

How it Works:

The failpoint make sure sending to  TiKV is not possible, whereas pessimistic transaction on a global temporary table can still read / write (involving pessimistic lock).

### Check List 

Tests 

- [ ] Unit test

### Release note 

- No release note

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -8456,7 +8417,14 @@ func (s testSerialSuite) TestTemporaryTableNoNetwork(c *C) {
 	// Index lookup
 	tk.HasPlan("select id from tmp_t where a = 1", "IndexLookUp")
 	tk.MustQuery("select id from tmp_t where a = 1").Check(testkit.Rows("1"))
+	tk.MustExec("rollback")
 
+	// Pessimistic lock
+	tk.MustExec("begin pessimistic")
```
Comment (lcwangchao) [inline executor/executor_test.go:8423]: It seems that a failed case will cause retry and makes test hanging for a long time

**Review by djshow832 — APPROVED** [review]

**Review by djshow832 — APPROVED** [review]

**Comment by tiancaiamao** [comment]
PTAL @coocood

**Review by coocood — APPROVED** [review]
LGTM

**Pull request merged**